### PR TITLE
added migration and test

### PR DESCRIPTION
**Pull request by Zartaj0**

### CODE REVIEW

File: contracts/PushCore/PushCoreV3.sol [hunk 1]
```diff
@@ -915,4 +915,14 @@ contract PushCoreV3 is
 
         emit ArbitraryRequest(sender, recipient, amount, feePercentage, feeID);
     }
+
+    function migaretAddresToBytes32(address[] calldata _channels) external whenPaused {
```
Comment (0xNilesh) [inline contracts/PushCore/PushCoreV3.sol:919]: migrateAddressToBytes32 typo :)

File: test/PushCore/unit_tests/ChannelCreationPush/ChannelMigration.t.sol [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            0
+        );
+        uint256 expectedProtocolPoolFees = FEE_AMOUNT * 2;
+        uint256 expectedChannelPoolFunds = (ADD_CHANNEL_MIN_FEES +
+            (ADD_CHANNEL_MIN_FEES * 2)) - expectedProtocolPoolFees;
+        assertEq(expectedProtocolPoolFees, coreV2.PROTOCOL_POOL_FEES());
+        assertEq(expectedChannelPoolFunds, coreV2.CHANNEL_POOL_FUNDS());
+    }
+
+    function test_migration() external {
```
Comment (0xNilesh) [inline test/PushCore/unit_tests/ChannelCreationPush/ChannelMigration.t.sol:72]: The name could be improved: test_ChannelsState_Migration

File: test/PushCore/unit_tests/ChannelCreationPush/ChannelMigration.t.sol [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        approveTokens(actor.governance, address(coreV2), 50_000 ether);
+        approveTokens(actor.bob_channel_owner, address(coreV2), 50_000 ether);
+        approveTokens(actor.alice_channel_owner, address(coreV2), 50_000 ether);
+        approveTokens(actor.charlie_channel_owner, address(coreV2), 50_000 ether);
+        approveTokens(actor.dan_push_holder, address(coreV2), 50_000 ether);
+        approveTokens(actor.tim_push_holder, address(coreV2), 50_000 ether);  
+        
+    }
+
+    function test_ProtocolPoolFeesCorrectForMultipleChannelsCreation()
```
Comment (0xNilesh) [inline test/PushCore/unit_tests/ChannelCreationPush/ChannelMigration.t.sol:47]: The name could be improved here as well: test_ProtocolPoolFees_IsCorrect_ForMultipleChannelsCreation

**Review by 0xNilesh — COMMENTED** [review]
Functionality wise LGTM!

**Review by 0xNilesh — APPROVED** [review]

File: test/PushCore/unit_tests/ChannelCreationPush/ChannelMigration.t.sol [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            address(coreV3)
+        );
+        coreV3 = PushCoreV3(address(epnsCoreProxyV2));
+        
+        bytes32[] memory _channelsBytes = new bytes32[](2);
+        _channelsBytes[0] = channelCreators.bob_channel_owner_Bytes32;
+        _channelsBytes[1] = channelCreators.charlie_channel_owner_Bytes32;
+
+        vm.expectRevert();
+        coreV3.migaretAddresToBytes32(_channels);
```
Comment (0xNilesh) [inline test/PushCore/unit_tests/ChannelCreationPush/ChannelMigration.t.sol:112]: this is remaining @Zartaj0

**Review by 0xNilesh — CHANGES_REQUESTED** [review]

**Review by 0xNilesh — APPROVED** [review]

**Review by zaryab2000 — APPROVED** [review]
Approving the PR LGTM

Just made one change to shift ChannelMigration test to its own folder, to maintain the structure of the test files. @Zartaj0

**Pull request merged**